app/assets/javascripts/code/services/_sensors.js

```diff
@@ -93,38 +93,39 @@ export const sensors = (params, $http) => {
       if (sensor) {
         return _(this.availableParameters).find(function(parameter) { return (parameter.id == sensor["measurement_type"]) });
       } else {
-        return null;
+        return ALL_PARAMETER;
       }
     },
     onSelectedParameterChange: function(selectedParameter, oldValue) {
       console.log('onSelectedParameterChange() - ', selectedParameter)
       if (selectedParameter === oldValue) return; // first angular watch run
       params.update({selectedSessionIds: []});
-      if (hasChangedToAll(selectedParameter)) {
-        this.availableSensors = sort(Object.values(this.sensors));
-        params.update({data: {sensorId: ""}});
+      if (selectedParameter.id === ALL_PARAMETER.id) {
+        this.availableSensors = [ALL_SENSOR].concat(sort(Object.values(this.sensors)));
+        params.update({data: {sensorId: ALL_SENSOR.id}});
       } else {
-        this.availableSensors = sort(Object.values(this.sensors).filter(sensor => sensor.measurement_type === selectedParameter.id));
+        this.availableSensors = [ALL_SENSOR].concat(sort(Object.values(this.sensors).filter(sensor => sensor.measurement_type === selectedParameter.id)));
```

Comment:
this `[ALL_SENSOR].concat(` is repeated in a few places. Maybe it would make sense to extract it into a function like `addAllSensor` or something.

Reply:
For now I prefer to leave the duplication. Not sure how this will evolve and prefer the right concretion vs an abstraction